Source/JavaScriptCore/wasm/WasmAirIRGeneratorBase.h

```diff
@@ -993,22 +989,15 @@ AirIRGeneratorBase<Derived, ExpressionType>::AirIRGeneratorBase(const ModuleInfo
     m_rootBlock = m_currentBlock;
 
     // FIXME we don't really need to pin registers here if there's no memory. It makes wasm -> wasm thunks simpler for now. https://bugs.webkit.org/show_bug.cgi?id=166623
-    const PinnedRegisterInfo& pinnedRegs = PinnedRegisterInfo::get();
-    m_wasmContextInstanceGPR = pinnedRegs.wasmContextInstancePointer;
-    m_code.pinRegister(m_wasmContextInstanceGPR);
+    m_code.pinRegister(GPRInfo::wasmContextInstancePointer);
 
     if constexpr (Derived::supportsPinnedStateRegisters) {
-
-        m_memoryBaseGPR = pinnedRegs.baseMemoryPointer;
-        m_code.pinRegister(m_memoryBaseGPR);
-
-        if (mode == MemoryMode::BoundsChecking) {
-            m_boundsCheckingSizeGPR = pinnedRegs.boundsCheckingSizeRegister;
-            m_code.pinRegister(m_boundsCheckingSizeGPR);
-        }
+        m_code.pinRegister(GPRInfo::wasmBaseMemoryPointer);
+        if (mode == MemoryMode::BoundsChecking)
+            m_code.pinRegister(GPRInfo::wasmBoundsCheckingSizeRegister);
     } else {
-        ASSERT(InvalidGPRReg == pinnedRegs.baseMemoryPointer);
-        ASSERT(InvalidGPRReg == pinnedRegs.boundsCheckingSizeRegister);
+        ASSERT(InvalidGPRReg == GPRInfo::wasmBaseMemoryPointer);
+        ASSERT(InvalidGPRReg == GPRInfo::wasmBoundsCheckingSizeRegister);
```

Comment:
`static_assert`.

Reply:
This does not work since it becomes compile error. if-constexpr isn't simple compile-time-if, and in this case, it will cause failure.